cores/esp8266/LwipIntfDev.h

```diff
@@ -272,22 +273,43 @@ boolean LwipIntfDev<RawDev>::begin(const uint8_t* macAddress, const uint16_t mtu
         }
     }
 
-    if (_intrPin < 0
-        && !schedule_recurrent_function_us(
+    static bool _scheduled = false;
+
+    if (_intrPin < 0 && !_scheduled)
+    {
+        _scheduled = schedule_recurrent_function_us(
             [&]()
             {
                 this->handlePackets();
+                if (!_started)
+                {
```

Comment:
You can use `_started` in place of `_scheduled` which become useless.
Inside, `handlePackets()` should be called *after* checking that scheduled function must be removed.

Reply:
if `begin` is executed immediately after `end`, `started` changes to true before the scheduled function is removed. the function is then scheduled again and the previous schedule is not removed.

with `_scheduled` the function stays scheduled if begin is executed before `_scheduled` changes to false. the schedule is removed if it executes before `begin` is invoked from the sketch.

Reply:
Right, with `_started` alone there would be a risk of having two scheduled functions when `begin()` is immediately called after `end()`.

There is still an issue with `_scheduled` because it is static. This prevents to use more than one instance.

And `handlePackets()` must be called *after* the `(!_started)` otherwise it would be called after `Rawdev::end()` is called.